rust/lance-index/src/scalar/inverted/index.rs

```diff
@@ -229,17 +229,38 @@ impl ScalarIndex for InvertedIndex {
     where
         Self: Sized,
     {
-        let token_reader = store.open_index_file(TOKENS_FILE).await?;
-        let invert_list_reader = store.open_index_file(INVERT_LIST_FILE).await?;
-        let docs_reader = store.open_index_file(DOCS_FILE).await?;
+        let tokens_fut = tokio::spawn({
+            let store = store.clone();
+            async move {
+                let token_reader = store.open_index_file(TOKENS_FILE).await?;
+                let tokens = TokenSet::load(token_reader).await?;
+                Result::Ok(tokens)
+            }
+        });
+        let invert_list_fut = tokio::spawn({
+            let store = store.clone();
+            async move {
+                let invert_list_reader = store.open_index_file(INVERT_LIST_FILE).await?;
+                let invert_list = InvertedListReader::new(invert_list_reader)?;
+                Result::Ok(Arc::new(invert_list))
+            }
+        });
+        let docs_fut = tokio::spawn({
+            let store = store.clone();
+            async move {
+                let docs_reader = store.open_index_file(DOCS_FILE).await?;
+                let docs = DocSet::load(docs_reader).await?;
+                Result::Ok(docs)
+            }
+        });
```

Comment:
It seems like we shouldn't need to spawn these.
```suggestion
        let tokens_fut = store.open_index_file(TOKENS_FILE)
            .and_then(|token_reader| TokenSet::load(token_reader));
        let invert_list_fut = store.open_index_file(INVERT_LIST_FILE)
            .and_then(|invert_list_reader| InvertedListReader::new(invert_list_reader))
            .map_ok(Arc::new);
        let docs_fut = store.open_index_file(DOCS_FILE)
            .and_then(|docs_reader| DocSet::load(docs_reader));
```